### PR TITLE
[MRG]Refactor task modules

### LINKED ISSUE #247: Refactor the .py file structures

### PR DESCRIPTION
**Pull request by tl-yang**
This resolves #247

#### PR detail
- Remove unused code in image_supervised and text_supervised. 
- Change inheritance structure of text_supervised.
- Move utility functions into utils.py
- Re-Use DataTransformer in TextClassifier. 

#### TODO 
- ImageClassfier and ImageRegressor should be saved into files with a different name instead of "classifier" for all.
- DataTransformer should be able to handle augmentation for both image and text(Currently no augmentation for text.
- export_keras_model, export_autokeras_model, load_searcher should be in supervised.

### CODE REVIEW

**Comment by jhfjhfj1** [comment]
@tl-yang Please resolve the conflicts. Is the validate_xy renamed? Thanks.

**Comment by tl-yang** [comment]
DataTransformer Architecture:

                DataTransformer(Abstract)
                    /           \
    ImageDataTransformer     TextDataTransformer

**Comment by jhfjhfj1** [comment]
This resolves #247

File: autokeras/text/text_supervised.py [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-from autokeras.preprocessor import OneHotEncoder
-from autokeras.search import Searcher
+from autokeras.preprocessor import OneHotEncoder, DataTransformer
+from autokeras.supervised import Supervised
 from autokeras.text.text_preprocessor import text_preprocess
-from autokeras.utils import pickle_to_file
+from autokeras.utils import pickle_to_file, validate_xy, temp_folder_generator, has_file, pickle_from_file
 
 
 class TextDataset(Dataset):
```
[the remark this answers is not in the document]
Comment (tl-yang) [reply autokeras/text/text_supervised.py]: Folder Strcuture: 
    nn
      loss
      metric
      model_trainer
      graph
      layer_transformer

    image
      image_supervised
      gan
    text
